### DISCUSSION TITLE
Removing demo data

### DISCUSSION THREAD

**Discussion by Neurocranium** [ghostfolio/ghostfolio#1670] [Q&A] [answered]
Hi community,

is there a way to remove the demo data from my self-hosted (docker/podman based) ghostfolio instance? After all, it is internally just for me. No need to have anybody around beside the admin and my day-to-day user.

I already tried to delete the data manually form the Postgres DB but then it just gets staged back in once I recreate my containers.

Best regards,
Neuro

**Comment by dtslvr** [comment]
The demo data originate from the database seed. You can't control this step.

I have also considered removing the demo account completely. Would be interesting to hear what others think about it.

**Reply by Neurocranium** [reply]
Thanks for that input. From my end, I would at least like to see an environment variable or something to be able to switch that on or off.

For the moment, the seed comes from this file here, right: `prisma/seed.js` What about I mount a stripped down version of this into my container? Any chance of breaking things in other places unless I provide a syntactical wrong js here?

**Reply by dtslvr** [reply]
Yes, the data is from `prisma/seed.js` triggered via `start:prod` in [package.json](https://github.com/ghostfolio/ghostfolio/blob/main/package.json)) from [Dockerfile](https://github.com/ghostfolio/ghostfolio/blob/main/Dockerfile).

Caution: not only the demo account is created, but also necessary data like the `EMERGENCY_FUND` tag.

**Reply by Neurocranium** [reply]
Worked like a charm 👍 

Here is what I did to clean out the demo data:

1. Stopped my ghostfolio app container
2. Entered the postgres db container and executed `psql -U myDBUser -d ghostfolio-db`
3. Executed the following DB commands to clear data from previous seeding
```
DELETE FROM "Account" WHERE "userId"='9b112b4d-3b7d-4bad-9bdd-3b0f7b4dac2f';
DELETE FROM "Order" WHERE "userId"='9b112b4d-3b7d-4bad-9bdd-3b0f7b4dac2f';
DELETE FROM "User" WHERE "id"='9b112b4d-3b7d-4bad-9bdd-3b0f7b4dac2f';
# ! Careful with the next two. They matched my pretty empty database but are NOT universal !
DELETE FROM "MarketData" WHERE "symbol"!='DZ7X.F';
DELETE FROM "SymbolProfile" WHERE "symbol"!='DZ7X.F';
```
4. Added the following stripped down `nodemo-seed.js` file to my host
```javascript
const {
  AccountType,
  DataSource,
  PrismaClient,
  Role,
  Type
} = require('@prisma/client');
const prisma = new PrismaClient();

async function main() {
  const platformBitcoinSuisse = await prisma.platform.upsert({
    create: {
      id: '70b6e475-a2b9-4527-99db-943e4f38ce45',
      name: 'Bitcoin Suisse',
      url: 'https://www.bitcoinsuisse.com'
    },
    update: {},
    where: { id: '70b6e475-a2b9-4527-99db-943e4f38ce45' }
  });

  const platformBitpanda = await prisma.platform.upsert({
    create: {
      id: 'debf9110-498f-4811-b972-7ebbd317e730',
      name: 'Bitpanda',
      url: 'https://www.bitpanda.com'
    },
    update: {},
    where: { id: 'debf9110-498f-4811-b972-7ebbd317e730' }
  });

  const platformCoinbase = await prisma.platform.upsert({
    create: {
      id: '8dc24b88-bb92-4152-af25-fe6a31643e26',
      name: 'Coinbase',
      url: 'https://www.coinbase.com'
    },
    update: {},
    where: { id: '8dc24b88-bb92-4152-af25-fe6a31643e26' }
  });

  const platformDegiro = await prisma.platform.upsert({
    create: {
      id: '94c1a2f4-a666-47be-84cd-4c8952e74c81',
      name: 'DEGIRO',
      url: 'https://www.degiro.eu'
    },
    update: {},
    where: { id: '94c1a2f4-a666-47be-84cd-4c8952e74c81' }
  });

  const platformInteractiveBrokers = await prisma.platform.upsert({
    create: {
      id: '9da3a8a7-4795-43e3-a6db-ccb914189737',
      name: 'Interactive Brokers',
      url: 'https://www.interactivebrokers.com'
    },
    update: {},
    where: { id: '9da3a8a7-4795-43e3-a6db-ccb914189737' }
  });

  const platformPostFinance = await prisma.platform.upsert({
    create: {
      id: '5377d9df-0d25-42c2-9d9b-e4c63166281e',
      name: 'PostFinance',
      url: 'https://www.postfinance.ch'
    },
    update: {},
    where: { id: '5377d9df-0d25-42c2-9d9b-e4c63166281e' }
  });

  const platformSwissquote = await prisma.platform.upsert({
    create: {
      id: '1377d9df-0d25-42c2-9d9b-e4c63156291f',
      name: 'Swissquote',
      url: 'https://swissquote.com'
    },
    update: {},
    where: { id: '1377d9df-0d25-42c2-9d9b-e4c63156291f' }
  });

  await prisma.tag.createMany({
    data: [
      {
        id: '4452656d-9fa4-4bd0-ba38-70492e31d180',
        name: 'EMERGENCY_FUND'
      }
    ],
    skipDuplicates: true
  });

  console.log({
    platformBitcoinSuisse,
    platformBitpanda,
    platformCoinbase,
    platformDegiro,
    platformInteractiveBrokers,
    platformPostFinance,
    platformSwissquote
  });
}

main()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

```
5. Afterwards added the mapping for the customized `nodemo-seed.js` to my container (podman pods instead of docker here):

`podman create --restart=always --pod=ghostfolio --name=ghost-app --env-file /opt/containers/ghostfolio/envs/ghostfolio.env -v /opt/containers/ghostfolio/customization/nodemo-seed.js:/ghostfolio/apps/api/prisma/seed.js:ro docker.io/ghostfolio/ghostfolio:latest`

Hope this helps others as well. :-) I guess this method could also be used to introduce a bit more customization like my own tags or platforms.

**Reply by dtslvr** [reply]
> I have also considered removing the demo account completely. Would be interesting to hear what others think about it.

Simplification of the database seeding is in progress: #1763

**Comment by Neurocranium** [comment, marked as answer]
Workaround with some `seed.js` customization in the [comment below](https://github.com/ghostfolio/ghostfolio/discussions/1670#discussioncomment-4898763).
